server/src/main/java/org/elasticsearch/index/mapper/Mapping.java

```diff
@@ -85,6 +87,33 @@ RootObjectMapper getRoot() {
         return root;
     }
 
+    /**
+     * Visit each defined Mapper and their children in the mapper tree
+     * @param consumer a callback that will receive each Mapper
+     */
+    public void forEachMapper(Consumer<Mapper> consumer) {
```

Comment:
If you look at how this is used when building MappingLookup, we sometimes want to differentiate between the different types of mapper - there it builds three different lists.  It's also a bit of a pain to build a flat list out of a tree structure as you end up either building lots of short-lived lists and then concatenating them, or passing a list to be populated into the top method, in which case you may as well generalise it to a Consumer.

It's better than `fieldMappers()` because it's directly on the Mapping where it belongs :)

Reply:
ok sounds good to me.